File: AalsiTrader/Models/TradingRules.swift
import Foundation

struct TradingConfig: Codable {
    var startingCapital: Double?
    var maxRiskPerTradePct: Double?
    var dailyLossLimitPct: Double?
    var maxPositions: Int?
    var maxSectorExposurePct: Double?
    var rsiOversoldThreshold: Double?
    var rsiOverboughtThreshold: Double?
    var minRewardRiskRatio: Double?
    var minTimeframeConfidence: Double?
    var rejectHighFalseBreakout: Bool?
    var requireAgentAlignment: Bool?
    var maxTradeDurationHours: Int?
    var exitOnMomentumExhaustion: Bool?
    var exitOnReversalSignal: Bool?
    var intradayExitTime: String?
    var maxSwingHoldingDays: Int?
    var hedgeEnabled: Bool?
    var brokeragePerOrder: Double?
}

struct TradingRules: Codable {
    let userId: String?
    var entryRules: [String]?
    var exitRules: [String]?
    var riskRules: [String]?
    var config: TradingConfig?
    var lastUpdated: String?
    var updatedBy: String?
}

File: AalsiTraderAndroid/app/src/main/java/com/aalsitrader/android/model/TradingRules.kt
package com.aalsitrader.android.model

import kotlinx.serialization.Serializable

@Serializable
data class TradingConfig(
    val startingCapital: Double? = null,
    val maxRiskPerTradePct: Double? = null,
    val dailyLossLimitPct: Double? = null,
    val maxPositions: Int? = null,
    val maxSectorExposurePct: Double? = null,
    val rsiOversoldThreshold: Double? = null,
    val rsiOverboughtThreshold: Double? = null,
    val minRewardRiskRatio: Double? = null,
    val minTimeframeConfidence: Double? = null,
    val rejectHighFalseBreakout: Boolean? = null,
    val requireAgentAlignment: Boolean? = null,
    val maxTradeDurationHours: Int? = null,
    val exitOnMomentumExhaustion: Boolean? = null,
    val exitOnReversalSignal: Boolean? = null,
    val intradayExitTime: String? = null,
    val maxSwingHoldingDays: Int? = null,
    val hedgeEnabled: Boolean? = null,
    val brokeragePerOrder: Double? = null,
)

@Serializable
data class TradingRules(
    val userId: String? = null,
    val entryRules: List<String>? = null,
    val exitRules: List<String>? = null,
    val riskRules: List<String>? = null,
    val config: TradingConfig? = null,
    val lastUpdated: String? = null,
    val updatedBy: String? = null,
)
